### DISCUSSION TITLE
How can AES/Rijndael MixColumns be implemented?

### DISCUSSION THREAD

**Discussion by isidroas** [mhostetter/galois#562] [Q&A]
I have tried this:

``` python
>>> GF = galois.GF(2**8)
>>> poly = galois.Poly([3, 1, 1, 2], field=GF)
>>> poly2 = galois.Poly([219, 19, 83, 69], field=GF)
>>> poly * poly2
Poly(112x^6 + 238x^5 + 61x^4 + 36x^3 + 48x^2 + 227x + 138, GF(2^8))
```

The `poly2` is the same than in [wikipedia test vectors](https://en.wikipedia.org/wiki/Rijndael_MixColumns#Test_vectors_for_MixColumn())

The expected output is `[142 77 161 188]`

**Comment by mhostetter** [comment]
Where are you getting that expected output from? You are multiplying two polynomials with degree 3. Therefore the result is necessarily degree 6. Your "expected output" has degree 3. I question the test vector.

**Reply by isidroas** [reply]
Don't look at my Python snippet because I have no idea of the theory. 

As for the test vectors, it seems that wikipedia changed the data since I published the question. Now they are: 
![2025-03-22-191505_](https://github.com/user-attachments/assets/5b085493-abcf-4998-b6d8-92e934c50276)

I would like to replicate one of these with your library.

**Reply by mhostetter** [reply]
`MixColumns()` is a function that permutes (re-orders) the elements. I don't see how multiplying two polynomials does that. You'll have to understand the theory of AES more to accomplish that.

**Reply by isidroas** [reply]
I will review the theory and then maybe I'll ask another question. Thanks
